### DISCUSSION TITLE
RabbitMQ trace

### DISCUSSION THREAD

**Discussion by Toxa45** [apache/skywalking#11043] [General]
Good afternoon.
I am using org.springframework.cloud:spring-cloud-starter-stream-rabbit , can you tell me how to enable traceId logging for rabbitMQ? Otherwise, when processing a message in the logs, I see TID=N/A and there is no tracing in skywalking.

**Comment by Toxa45** [comment]
An example of a listener that does not log traceId
```
@Slf4j
@Configuration
@RequiredArgsConstructor
public class MessageListener {
    private final MessageProcessor messageProcessor;
    @Bean 
    public Consumer<StatusChangedData> statusChanged() {
        return messageProcessor::statusChanged;
    }
}

@Slf4j
@Service
@RequiredArgsConstructor
@Validated
@FieldDefaults(level = AccessLevel.PRIVATE, makeFinal = true)
public class MessageProcessor {
		
		public void statusChanged(@Valid StatusChangedData event) {
			log.info("event: {}", event);
		}
}
```

**Reply by wu-sheng** [reply]
You could find sample codes in https://github.com/apache/skywalking-java/tree/main/test/plugin/scenarios

**Reply by wu-sheng** [reply]
If this doesn't match your case, you have to check plugin source codes on your own.

**Comment by Toxa45** [comment]
thanks, I will take a look
